engine/src/api/endpoints/agent.py

```
@@ -546,3 +547,63 @@ async def stop_run(request: Request, user=Depends(fastapi_users.current_user(opt
+                name=name,
+                title=title,
+                annotations=annotations,
+                tags=tags
+            )
+            formatted_tools.append(formatted_tool)
+        
+        return formatted_tools
+
+    except Exception as e:
+        logger.exception(f"Error listing tools: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Error listing tools") from e
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)